clients/pkg/promtail/scrapeconfig/scrapeconfig.go

```diff
@@ -309,6 +310,27 @@ type GelfTargetConfig struct {
 	UseIncomingTimestamp bool `yaml:"use_incoming_timestamp"`
 }
 
+type CloudflareConfig struct {
+	// APIToken is the API key for the Cloudflare account.
+	APIToken string `yaml:"api_token"`
+	// ZoneID is the ID of the zone to use.
+	ZoneID string `yaml:"zone_id"`
+	// Labels optionally holds labels to associate with each record read from cloudflare logs.
+	Labels model.LabelSet `yaml:"labels"`
+	// The amount of workers to use for parsing cloudflare logs. Default to 3.
+	Workers int `yaml:"workers"`
+	// The timerange to fetch for each pull request that will be spread across workers. Default 1m.
+	PullRange model.Duration `yaml:"pull_range"`
```

Comment:
To clarify the purpose of this: are there expected to be so many logs from CloudFlare that one goroutine could not keep up? If so, we should probably expose a metric to measure that backpressure.

Reply:
There's a metric already. `cloudflare_target_last_end_timestamp`.

For most people it will be fine, for very high throughput, above 10k/s I expect people to require tweaking this.

Reply:
OK thanks. How would an operator distinguish between backpressure on log ingestion vs no more logs available in CloudFlare with this metric?

Reply:
If no logs are being received this timestamp still move since it's not the last received timestamp but the last pull end timestamp

Reply:
OK, I think we should make that very clear from the metric name and the description; I would've read this (and I suggested renaming to) "highest timestamp received", but it's not that at all.

Reply:
Previously it was `cloudflare_target_last_end_timestamp` would `cloudflare_target_last_requested_end_timestamp` do the job.

Reply:
I think that's pretty clear, yeah thanks :+1: 

Maybe it's useful to add a metric for the highest received timestamp too, which will make a calculation of how far promtail is behind quite simple like we do for remote-write:

```promql
prometheus_remote_storage_highest_timestamp_in_seconds
- prometheus_remote_storage_queue_highest_sent_timestamp_seconds
```

I'm not aware of a way to get the highest timestamp from a series in PromQL/LogQL - if there is, we can just use that.

Reply:
I suppose `time() - cloudflare_target_last_requested_end_timestamp` would probably be close enough, nevermind

Reply:
Exactly I use that for knowing the compactor lag